src/pkg/services/m365/api/users.go

```diff
@@ -268,6 +269,18 @@ func EvaluateMailboxError(err error) error {
 	return err
 }
 
+// IsErrMailboxNotFound inspects the secondary errors inside MailboxInfo and
+// determines whether the resource has a mailbox.
+func IsErrMailboxNotFound(errs []error) bool {
```

Comment:
Nit: `isAny` to imply that it will return true if any error matches.
```suggestion
func IsAnyErrMailboxNotFound(errs []error) bool {
```